crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -631,34 +632,78 @@ fn ensure_quote_accuracy(
-                - jit_orders_executed_amounts
+            let jit_orders_net_token_changes = trade.jit_orders_net_token_changes()?;
+            let jit_orders_sell_token_changes = jit_orders_net_token_changes
+                .get(&query.sell_token)
+                .context("missing jit orders sell token executed amount")?;
+            let jit_orders_buy_token_changes =
+                jit_orders_net_token_changes
                     .get(&query.buy_token)
                     .context("missing jit orders buy token executed amount")?;
-            (expected_sell_token_lost, expected_buy_token_lost)
+            // sell-sell
+            // jit_orders_sell_token_changes: -400.0
+            // jit_orders_buy_token_changes: 200.0
+            // sell_amount: 500.0
+            // buy_amount: 250.0
+
+            // buy-buy
+            // jit_orders_sell_token_changes: 400.0
+            // jit_orders_buy_token_changes: -200.0
+            // sell_amount: 500.0
+            // buy_amount: 250.0
+
+            // buy-sell
+            // jit_orders_sell_token_changes: -400.0
+            // jit_orders_buy_token_changes: 200.0
+            // sell_amount: 500.0
+            // buy_amount: 250.0
+
+            // sell-buy
+            // jit_orders_sell_token_changes: 400.0
+            // jit_orders_buy_token_changes: -200.0
+            // sell_amount: 500.0
+            // buy_amount: 250.0
+
+            match query.kind {
+                OrderKind::Sell => {
+                    let expected_sell_token_lost = -&sell_amount - jit_orders_sell_token_changes;
+                    let expected_buy_token_lost = &buy_amount - jit_orders_buy_token_changes;
+                    (expected_sell_token_lost, expected_buy_token_lost)
+                }
+                OrderKind::Buy => {
+                    let expected_sell_token_lost = -(&sell_amount - jit_orders_sell_token_changes);
+                    let expected_buy_token_lost = &buy_amount + jit_orders_buy_token_changes;
+                    (expected_sell_token_lost, expected_buy_token_lost)
+                }
+            }
         }
-        TradeKind::Legacy(_) => (sell_amount, buy_amount),
+        TradeKind::Legacy(_) => (BigRational::zero(), BigRational::zero()),
     };
 
-    if summary.sell_tokens_lost
-        >= inaccuracy_limit * u256_to_big_rational(&expected_sell_token_lost)
-        || summary.buy_tokens_lost
-            >= inaccuracy_limit * u256_to_big_rational(&expected_buy_token_lost)
-    {
+    let sell_token_lost = (&summary.sell_tokens_lost - expected_sell_token_lost)
+        .abs()
+        .to_f64()
+        .unwrap();
+    let sell_token_lost_limit = (inaccuracy_limit * &sell_amount).to_f64().unwrap();
+    let buy_token_lost = (&summary.buy_tokens_lost - expected_buy_token_lost)
+        .abs()
+        .to_f64()
+        .unwrap();
+    let buy_token_lost_limit = (inaccuracy_limit * &buy_amount).to_f64().unwrap();
```

Comment:
The following code returns false: 
```
BigRational::from_u64(50) == (BigRational::from_float(0.1) * BigRational::from_u64(500))
```
Diving deeper, BigRational value on the right side is something like `50.0000000000000349384` making the assertion fail. While converting to f64 works fine 🤷‍♂️ Already tried reducing the value with no luck.